File: 0009-palindrome-number/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x < 0) // signed values aren't palindromes
        return false;
    
    var num = x; // copy of x for calculation
    var rev = 0; // reversed value holder
    
    while(num != 0) {
        rev = (rev * 10) + (num % 10);
        num = Math.floor(num / 10); // integer division
    }
    
    return rev === x;
};

File: 0009-palindrome-number/0009-palindrome-number.go
func isPalindrome(x int) bool {
    if x < 0 { // can't be palindrome if signed
        return false;
    }
    
    var num = x; // copy x value
    var reversed int = 0; 
    for num != 0 {
        reversed = (reversed * 10) + (num % 10)
        num /= 10;
    }
    
    return reversed == x;
}